src/Microsoft.VisualBasic.Core/tests/StringTypeTests.cs

```diff
@@ -0,0 +1,386 @@
+        {
+            yield return new object[] { new object() };
+        }
+
+        private static IEnumerable<object[]> FromSingle_Format_TestData()
+        {
+            yield return new object[] { (float)(-1), default(NumberFormatInfo), "-1" };
+            yield return new object[] { (float)(-1), new NumberFormatInfo() { NegativeSign = "#" }, "#1" };
+        }
+
+        private static IEnumerable<object[]> FromDouble_Format_TestData()
+        {
+            yield return new object[] { (double)(-1), default(NumberFormatInfo), "-1" };
+            yield return new object[] { (double)(-1), new NumberFormatInfo() { NegativeSign = "#" }, "#1" };
+        }
+
+        private static IEnumerable<object[]> FromDecimal_Format_TestData()
+        {
+            yield return new object[] { (decimal)(-1), default(NumberFormatInfo), "-1" };
+            yield return new object[] { (decimal)(-1), new NumberFormatInfo() { NegativeSign = "#" }, "#1" };
+        }
+
+        [Theory]
+        [InlineData("a", 1, 0, "", "a")]
+        [InlineData("a", 1, 1, "", "a")]
+        [InlineData("a", 1, 0, "b", "a")]
+        [InlineData("a", 1, 1, "b", "b")]
+        [InlineData("a", 1, 2, "b", "b")]
+        [InlineData("abc", 2, 0, "def", "abc")]
+        [InlineData("abc", 2, 1, "def", "adc")]
+        [InlineData("abc", 2, 2, "def", "ade")]
+        [InlineData("abc", 2, 3, "def", "ade")]
+        public void MidStmtStr(string str, int start, int length, string insert, string expected)
+        {
+            StringType.MidStmtStr(ref str, start, length, insert);
+            Assert.Equal(expected, str);
+        }
+
+        [Theory]
+        [InlineData("", -1, 0, "")]
+        [InlineData("", 0, 0, "")]
+        [InlineData("", 1, 0, "")]
+        [InlineData("", 2, 0, "")]
+        [InlineData("", 1, -1, "")]
+        [InlineData("abc", 4, 0, "")]
+        [InlineData("abc", 1, -3, "")]
+        public void MidStmtStr_ArgumentException(string str, int start, int length, string insert)
+        {
+            Assert.Throws<ArgumentException>(() => StringType.MidStmtStr(ref str, start, length, insert));
+        }
+
+        [Theory]
+        [InlineData("a", "a", 0, 0)]
+        [InlineData("a", "b", -1, -1)]
+        [InlineData("b", "a", 1, 1)]
+        [InlineData("a", "ABC", 32, -1)]
+        [InlineData("ABC", "a", -32, 1)]
+        [InlineData("abc", "ABC", 32, 0)]
+        public void StrCmp(string left, string right, int expectedBinaryCompare, int expectedTextCompare)
+        {
+            Assert.Equal(expectedBinaryCompare, StringType.StrCmp(left, right, TextCompare: false));
+            Assert.Equal(expectedTextCompare, StringType.StrCmp(left, right, TextCompare: true));
+        }
+
+        [Theory]
+        [InlineData("", "*", true, true)]
+        [InlineData("", "?", false, false)]
+        [InlineData("a", "?", true, true)]
+        [InlineData("a3", "[A-Z]#", false, true)]
+        [InlineData("A3", "[a-z]#", false, true)]
```

Comment:
can you add some test cases with more patterns and strings. Also tests for null inputs are missings here